### PR TITLE
mgis 2.0 (new formula)

### PR DESCRIPTION
**Pull request by maxence-wz**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/mgis.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+class Mgis < Formula
+  desc "Provide tools to handle MFront generic interface behaviours"
+  homepage "https://thelfer.github.io/mgis/web/index.html"
+  license "GPL-1.0-or-later"
+  head "https://github.com/thelfer/MFrontGenericInterfaceSupport.git", using: :git, branch: "master"
+
+  stable do
+    url "https://github.com/thelfer/MFrontGenericInterfaceSupport/archive/refs/tags/MFrontGenericInterfaceSupport-2.0.tar.gz"
+    sha256 "cb427d77f2c79423e969815b948a8b44da33a4370d1760e8c1e22a569f3585e2"
+    patch :DATA
```
Comment (SMillerDev) [inline Formula/mgis.rb:10]: Is there an upstream version of this?

Comment (chenrui333) [reply Formula/mgis.rb:10]: looks like it is [this commit](https://github.com/thelfer/MFrontGenericInterfaceSupport/commit/3861ac90c76d69956b7267a8d0cc663bdf2e3689), but the commit has some syntax issue, not being released yet.

File: Formula/mgis.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      "-Denable-portable-build=ON",
+      "-Denable-website=OFF",
+      "-Denable-enable-doxygen-doc=OFF",
+      "-Denable-c-bindings=ON",
+      "-Denable-fortran-bindings=ON",
+      "-Denable-python-bindings=ON",  # requires boost-python
+      "-Denable-fenics-bindings=OFF", # experimental and very limited
+      "-Denable-julia-bindings=OFF",  # requires CxxWrap library
+      "-Denable-enable-static=OFF",
+      "-Ddisable_python_library_linking=ON",
```
Comment (chenrui333) [inline Formula/mgis.rb:30]:
```suggestion
      "-Ddisable_python_library_linking=ON",
      "-DCMAKE_INSTALL_RPATH=#{rpath}",
```

**Comment by chenrui333** [comment]
@maxence-wz can you help create an upstream issue to push for a release for the build patch? Thanks!

**Comment by chenrui333** [comment]
```
  ==> python3 -c import mgis.behaviour > log_behaviour.out
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
  ModuleNotFoundError: No module named 'mgis'
  Error: mgis: failed
```

File: Formula/mgis.rb (hunk 1 above)
Comment (maxence-wz) [reply Formula/mgis.rb:9]: @SMillerDev, yes [here](https://github.com/thelfer/MFrontGenericInterfaceSupport/commit/bfeb0ab537ef2870914986d193854fe0d810d5d2)

**Comment by maxence-wz** [comment]
> @maxence-wz can you help create an upstream issue to push for a release for the build patch? Thanks!

Hi @chenrui333, thx for your help. What do you mean ? The patch modification has been integrated for the next stable release of mgis in [this](https://github.com/thelfer/MFrontGenericInterfaceSupport/commit/bfeb0ab537ef2870914986d193854fe0d810d5d2) commit.

File: Formula/mgis.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/mgis.rb:9]: Could we use that version then?

Comment (maxence-wz) [reply Formula/mgis.rb:9]: @SMillerDev I'm asking when it will be released

Comment (maxence-wz) [reply Formula/mgis.rb:9]: @SMillerDev no answer for the moment but the patch has been approved upstream

**Pull request closed without merging**